src/main/scala/com/cloudant/clouseau/IndexManagerService.scala

```diff
@@ -94,6 +95,21 @@ class IndexManagerService(ctx: ServiceContext[ConfigurationArgs]) extends Servic
   val openTimer = metrics.timer("opens")
   val lru = new LRU()
   val waiters = Map[String, List[(Pid, Any)]]()
+  val countLocksEnabled = ctx.args.config.getBoolean("clouseau.count_locks", false)
+  if (countLocksEnabled) {
+    metrics.gauge("NativeFSLock.count")(getNativeFSLockHeldSize)
+  }
+
+  def getNativeFSLockHeldSize(): Long = {
+    synchronized {
+      val lockClass = Class.forName("org.apache.lucene.store.NativeFSLock")
+      val field = lockClass.getDeclaredField("LOCK_HELD")
+      field.setAccessible(true)
+      val LOCK_HELD = field.get(null).asInstanceOf[HashSet[String]]
+      field.setAccessible(false)
+      LOCK_HELD.size
```

Comment:
we need to hold a lock on the LOCK_HELD object (like everything else that access it), but this is really close.

Reply:
Would this be an additional `synchronized` section?  Something like this:

```scala
  def getNativeFSLockHeldSize(): Long = {
    synchronized {
      val lockClass = Class.forName("org.apache.lucene.store.NativeFSLock")
      val field = lockClass.getDeclaredField("LOCK_HELD")
      field.setAccessible(true)
      val LOCK_HELD = field.get(null).asInstanceOf[HashSet[String]]
      field.setAccessible(false)
      LOCK_HELD.synchronized {
        LOCK_HELD.size
      }
    }
  }
```

Reply:
Ah, I managed to come up with a simplified solution.  I have also got reminded that `synchronized` is needed for establishing the "happens before" guarantee on concurrent reads.